x/ibc/02-client/types/errors/errors.go

```diff
@@ -1,4 +1,4 @@
-package types
+package errors
```

Comment:
I'm not against this change, but it breaks the standard [module structure](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).

 I like how @alexanderbez implemented the errors on the evidence module [here](https://github.com/cosmos/cosmos-sdk/pull/5240/files#diff-1190e89d7b69c54b0bb54995ef7c1da4R23).

Reply:
If i leave `errors.go` in `types/`, there wasn't a way for me to use those errors in the `types/tendermint` package without creating an import cycle. This is why i split it into its own package

Reply:
> The current ICS-02 implementation uses the old sdk.Error, should I switch it to use new error type?

Thoughts @alexanderbez ?

Reply:
So the approach I've taken in the evidence module PR and for future PRs is the following:

Return `error` where ever and as much as possible up and until I can no longer return an `error`. When I can no longer return an `error`, I call `sdk.ConvertError(err)` which gives me the old (soon to be deprecated) `sdk.Error`.

Behind the scenes I'm returning a concrete type of `sdkerrors.Error{...}` which fulfills the `error` interface and fulfills the ABCI semantics.

so tl;dr, return `error` using `sdkerrors.New(...)` as much as possible until you cant.